### PR TITLE
[NXCM-4895] Introduce SmtpSessionParametersCustomizer for allowing customisation of SMTP session parameters

### PR DESCRIPTION
**Pull request by adreghiciu**
Signed-off-by: Alin Dreghiciu adreghiciu@gmail.com

### CODE REVIEW

**Comment by adreghiciu** [comment]
CI build: https://bamboo.zion.sonatype.com/browse/NXO-OSSF20-1

File: nexus-core/src/main/java/org/sonatype/nexus/email/DefaultNexusEmailer.java [hunk 1]
```diff
@@ -55,6 +58,9 @@
     @Requirement
     private EMailer eMailer;
 
+    @Requirement
+    private PlexusContainer plexus;
```
Comment (nabcos) [inline nexus-core/src/main/java/org/sonatype/nexus/email/DefaultNexusEmailer.java]: better convert to `@Named` and use method injection or `Provider<EMailerConfiguration>`?

**Comment by ifedorenko** [comment]
I really dislike plexus.lookup/catch(ComponentLookupException). I think NexusEmailerConfiguration does not need to be a component, it should be plain class with package visibility.

Otherwise +1.

**Comment by adreghiciu** [comment]
CI :+1:

**Comment by jdillon** [comment]
-1

We should not be introducing anymore more use of PlexusContainer... we should _ONLY_ be removing usage of this.

**Comment by adreghiciu** [comment]
@ifedorenko Changed to use "new"

**Comment by adreghiciu** [comment]
@jdillon did not want to convert this yet to sisu, but maybe we should do it

**Comment by jdillon** [comment]
@adreghiciu if we come across simple plexus components when adding/fixing features we should try and convert them.  Or we'll never finish the transition from plexus to guice.

Anyways, my main beef here was with use of PlexusContainer directly.  We _MUST NOT_ use this anymore.

**Pull request merged**